Userland/Libraries/LibGfx/ICC/TagTypes.cpp

```diff
@@ -358,6 +358,17 @@ static ErrorOr<CLUTData> read_clut_data(ReadonlyBytes bytes, AdvancedLUTHeader c
     return CLUTData { move(number_of_grid_points_in_dimension), move(values) };
 }
 
+static ErrorOr<Vector<LutCurveType>> read_curves(ReadonlyBytes bytes, u32 offset, u32 count)
+{
+    Vector<LutCurveType> curves;
+    // FIXME: Implement.
+    (void)bytes;
+    (void)offset;
+    for (u32 i = 0; i < count; ++i)
+        TRY(curves.try_append(adopt_ref(*new CurveTagData(0, 0, {}))));
```

Comment:
For extra GBP, you can `adopt_nonnull_own_or_enomem(new (nothrow) CurveTagData...` and also fail with `ENOMEM` here.

Reply:
I'll make a separate pr to update this whole file to that.

Reply:
Hmm, looks like there isn't a `adopt_nonnull_ref_or_enomem`, and this uses RefPtrs, not OwnPtrs. Should there be something like that for RefPtrs too?

Reply:
Ah no, it exists, it's just in RefPtr.h instead of in NonnullRefPtr.h!

Reply:
-> #17426.

Reply:
…feel like updating Documentation/SmartPointers.md with when to pick `adopt_nonnull_own_or_enomem` over `adopt_own` (and same for RefPtr)? (I'm guessing it's "always" (?), but the doc currently doesn't mention these at all and recommend the old tech.)

Reply:
II opened #17432 for the documentation needing an update.